web/src/containers/OpenOrders/index.tsx

```diff
@@ -1,180 +1,164 @@
-}
+        dispatch(openOrdersCancelFetch({ order: orderToDelete, list }));
+    }, [list]);
 
-const mapStateToProps = (state: RootState): ReduxProps => ({
-    currentMarket: selectCurrentMarket(state),
-    list: selectOpenOrdersList(state),
-    fetching: selectOpenOrdersFetching(state),
-    cancelFetching: selectCancelOpenOrdersFetching(state),
-    userLoggedIn: selectUserLoggedIn(state),
-});
+    const handleCancelAll = useCallback(() => {
+        currentMarket && dispatch(ordersCancelAllFetch({ market: currentMarket.id }));
+    }, [currentMarket]);
 
-const mapDispatchToProps: MapDispatchToPropsFunction<DispatchProps, {}> = dispatch => ({
-    userOpenOrdersFetch: payload => dispatch(userOpenOrdersFetch(payload)),
-    openOrdersCancelFetch: payload => dispatch(openOrdersCancelFetch(payload)),
-    ordersCancelAll: payload => dispatch(ordersCancelAllFetch(payload)),
-});
-
-export type OpenOrdersProps = ReduxProps;
-
-export const OpenOrdersComponent = injectIntl(
-    connect(
-        mapStateToProps,
-        mapDispatchToProps,
-    )(OpenOrdersContainer),
-) as React.FunctionComponent;
+    const classNames = useMemo(() => 
+        classnames('pg-open-orders', {
+            'pg-open-orders--empty': !list.length,
+            'pg-open-orders--loading': fetching,
+        }),
+    [list, fetching]);
+
+    const handleToggleCheckbox = React.useCallback(event => {
+        event.preventDefault();
+        setHideOtherPairs(!hideOtherPairs);
+    }, [hideOtherPairs]);
+
+    return (
+        <div className={classNames}>
+            <div className="cr-table-header__content">
+                <div className="cr-title-component">
+                    <span>{translate('page.body.trade.header.openOrders')}</span>
+                    <Form className="cr-title-component__checkbox" onClick={handleToggleCheckbox}>
+                        <Form.Check
+                            type="checkbox"
+                            custom
+                            id="hideOtherPairs"
+                            checked={hideOtherPairs}
+                            readOnly={true}
+                            label={translate('page.body.trade.header.openOrders.hideOtherPairs')}
+                        />
+                    </Form>
+                    <span className="cr-table-header__cancel" onClick={handleCancelAll}>
+                        <FormattedMessage id="page.body.openOrders.header.button.cancelAll" />
+                        <CloseIcon className="cr-table-header__close" />
+                    </span>
+                </div>
+            </div>
+            {fetching && <div className="open-order-loading"><Spinner animation="border" variant="primary" /></div>}
+            {!fetching && (
+                <OpenOrders
+                    headersKeys={headersKeys}
+                    headers={renderHeaders}
+                    data={renderData}
+                    onCancel={handleCancel}
+                />
+            )}
```

Comment:
Please create a method using useMemo that returns html depends on `fetching` value